docs/book/src/migration/migration_guide_v2tov3.md

```diff
@@ -0,0 +1,239 @@
+
+```
+...
+projectName: example
+...
+```
+
+- Add the `layout`
+
+The default plugin layout which is equivalent to the previous versions is `go.kubebuilder.io/v2`: 
+
+```
+...
+layout: go.kubebuilder.io/v2
+...
+```
+
+- Add the `version`
+
+The `version` field represents the version of Project layouts. So, you ought to update this to `3-alpha`:
+
+```
+...
+version: 3-alpha`
+...
+```
+
+The final result of your project file would like:
+
+```sh
+domain: my.domain
+layout: go.kubebuilder.io/v2
+projectName: example
+repo: example
+resources:
+- group: webapp
+  kind: Guestbook
+  version: v1
+version: 3-alpha
+```
+
+### Verification
+
+Finally, we can run `make` and `make docker-build` to ensure things are working
+fine.
+
+## Migrating your projects to use v3+ plugins
+
+<aside class="note warning">
+
+<h1>Note</h1>
+
+v3+ plugins still in alpha version and are not scaffolded by default. You are able to use then and try it out and get all improvements made so far by initializing projects with the arg `--plugins=go.kubebuilder.io/v3-alpha`: 
+
+```sh 
+kubebuilder init --domain my.domain --plugins=go.kubebuilder.io/v3-alpha
+```
+
+</aside>
+
+<aside class="note warning">
+
+<h1>Note</h1>
+
+Currently, the plugin `v3-alpha` has NO breaking changes. However, until it get the stable state it might address some breaking changes to address some wished requirements related to k8s API deprecations of `CustomResourceDefinition` and `ValidatingWebhookConfiguration` `v1beta1` version as to upgrade the `cert-manager`. More info:
+
+- [cert-manager related configuration should be migrated to cert-manager.io/v1 #1666
+](https://github.com/kubernetes-sigs/kubebuilder/issues/1666)
+- [Set preserveUnknownFields to false in the CRD conversion webhook patch #933](https://github.com/kubernetes-sigs/kubebuilder/issues/933)
+- [Migrate existing KB project to v1 CRDs and Webhooks with minimal user effort #1065
+](https://github.com/kubernetes-sigs/kubebuilder/issues/1065)
+   
+</aside>
+
+### Prerequisites
+
+Following the above steps to migrate your project to
+v3 project layout.
+
+### Update your PROJECT file
+
+Update the `layout` with the new plugin version ` go.kubebuilder.io/v3-alpha` and it would like to: 
+
+```sh
+ $ cat PROJECT 
+domain: my.domain
+layout: go.kubebuilder.io/v3-alpha
+projectName: example
+repo: example
+resources:
+- group: webapp
+  kind: Guestbook
+  version: v1
+version: 3-alpha
+
+```
+
+### V3+ plugins significant changes in the scalffold
+
+<aside class="note warning">
+
+<h1>Note</h1>
+
+The following changes are NOT breaking changes. In this way, these changes are optional but recommended in order to keep your project updated with the latest changes made so far. 
+
+</aside>
+
+#### Manager Rootless
+
+The projects built with v3+ plugin now define a specific `user ID` and `securityContext` policy for not allow to escalate root privileges from the manager container. 
```

Comment:
```suggestion
The projects built with v3+ plugin now define a specific `user ID` and `securityContext` policy to prevent root privilege escalation from the manager container. 
```